src/main/java/org/openrewrite/java/testing/assertj/CollapseConsecutiveAssertThatStatements.java

```diff
@@ -0,0 +1,169 @@
+    private static final MethodMatcher ASSERT_THAT = new MethodMatcher("org.assertj.core.api.Assertions assertThat(..)");
+
+    @Override
+    public String getDisplayName() {
+        return "Collapse consecutive `assertThat` statements";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Collapse consecutive `assertThat` statements into single `assertThat` chained statement. This recipe ignores `assertThat` statements that have method invocation as parameter.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(new UsesType<>(ASSERTJ_QUALIFIED_ASSERTIONS_CLASS_NAME, true), new CollapseConsecutiveAssertThatStatementsVisitor());
+    }
+
+    private static class CollapseConsecutiveAssertThatStatementsVisitor extends JavaIsoVisitor<ExecutionContext> {
+        @Override
+        public J.Block visitBlock(J.Block _block, ExecutionContext ctx) {
+            J.Block block = super.visitBlock(_block, ctx);
+
+            List<List<J.MethodInvocation>> consecutiveAssertThatStatementList = getConsecutiveAssertThatList(block.getStatements());
+
+            List<Statement> modifiedStatements = new ArrayList<>();
+            int currIndex = 0;
+            while (currIndex < consecutiveAssertThatStatementList.size()) {
+                if (consecutiveAssertThatStatementList.get(currIndex).size() <= 1) {
+                    modifiedStatements.add(block.getStatements().get(currIndex));
+                    currIndex += 1;
+                } else {
+                    modifiedStatements.add(getCollapsedAssertThat(consecutiveAssertThatStatementList.get(currIndex)));
+                    currIndex += consecutiveAssertThatStatementList.get(currIndex).size();
+                }
+            }
+
+            block = block.withStatements(modifiedStatements);
+
+            return maybeAutoFormat(_block, block.withStatements(modifiedStatements), ctx);
+        }
+
+
+        private List<List<J.MethodInvocation>> getConsecutiveAssertThatList(List<Statement> statements) {
+            List<List<J.MethodInvocation>> consecutiveAssertThatList = new ArrayList<>();
+            String prevArg = "";
+            int currListIndex = 0;
+            int statementListSize = statements.size();
+            List<J.MethodInvocation> currList = new ArrayList<>();
+            for (int currIndex = 0; currIndex < statementListSize; currIndex++) {
+
+                consecutiveAssertThatList.add(new ArrayList<>());
+
+                Statement statement = statements.get(currIndex);
+                if (statement instanceof J.MethodInvocation) {
+                    Optional<J.MethodInvocation> assertThatMi = getAssertThatMi(statement);
+                    if (assertThatMi.isPresent()) {
+                        if (isAssertThatValid(statement)) {
+                            if (!getFirstArgumentName(assertThatMi.get()).equals(prevArg)) {
```

Comment:
Does it make sense to use org.openrewrite.java.search.SemanticallyEqual here instead of just checking the argument name?